test/extended/util/framework.go

```diff
@@ -280,3 +281,19 @@ func ExtendedTestPath() string {
 func FixturePath(elem ...string) string {
 	return filepath.Join(append([]string{ExtendedTestPath()}, elem...)...)
 }
+
+// CheckURL grabs the output from the specified url and returns it.
+// It will retry up to N times if an error occurs during the request.
+func CheckURL(url string, retries int) (response string, err error) {
+	r, err := http.Get(url)
+	count := 0
+	for count < retries && err != nil {
```

Comment:
wait.Poll returns just an error, it doesn't allow you to return content.

so i'd have to wait.Poll, then make the http request again afterwards to actually get the content.

or i'd have to change this function to actually do the checking for the content instead of just returning it, which makes it less generically useful.